.github/workflows/build-test-emscripten.yml

```
@@ -51,6 +51,20 @@ jobs:
             mkdir ${GITHUB_WORKSPACE}
           fi
 
+        # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
+      - name: Get AWS instance type
+        run: |
+          TOKEN=$(curl -fsS --max-time 2 \
+            -X PUT "http://169.254.169.254/latest/api/token" \
+            -H "X-aws-ec2-metadata-token-ttl-seconds: 10" || true)
+
+          AWS_INSTANCE_TYPE=$(curl -fsS --max-time 2 \
+            -H "X-aws-ec2-metadata-token: $TOKEN" \
+            "http://169.254.169.254/latest/meta-data/instance-type" || true)
+
```

```suggestion
    #  if not an EC2 instance
    if [[ -z "$AWS_INSTANCE_TYPE" ]]; then
      AWS_INSTANCE_TYPE="self-hosted"
    fi
```

we have a fleet of self-hosted arm runners, so mb add a fallback?